src/magic_folder/__init__.py

```diff
@@ -1,5 +1,9 @@
 # Copyright (C) Least Authority TFA GmbH
 
+# see https://github.com/LeastAuthority/magic-folder/issues/305
+import warnings
+warnings.filterwarnings("ignore", module="OpenSSL.crypto.*", lineno=14)
```

Comment:
I removed the cryptography version constraint in setup.py:

```diff
diff --git a/setup.py b/setup.py
index 5ce7ccd4..f1bfa3ed 100644
--- a/setup.py
+++ b/setup.py
@@ -71,7 +71,7 @@ install_requires = [
 
     # Tahoe-LAFS 1.14.0 happens to be incompatible with cryptography 3.0.
     # https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3349
-    "cryptography < 3.0",
+    "cryptography",
 
     "treq",
```

And then did an editable install of magic-folder into a python 2.7 virtualenv:

```sh
$ pip freeze
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support pip 21.0 will remove support for this function
ality.
appdirs==1.4.4
atomicwrites==1.4.0
attrs==19.3.0
autobahn==19.11.2
Automat==20.2.0
backports.functools-lru-cache==1.6.4
bcrypt==3.1.7
boltons==20.2.1
certifi==2020.12.5
cffi==1.14.5
chardet==4.0.0
click==7.1.2
click-default-group==1.2.2
configparser==4.0.2
constantly==15.1.0
contextlib2==0.6.0.post1
coverage==4.5.4
cryptography==3.3.2
decorator==4.4.2
distlib==0.3.1
distro==1.5.0
eliot==1.7.0
enum34==1.1.10
extras==1.0.0
filelock==3.0.12
fixtures==3.0.0
foolscap==0.13.1
funcsigs==1.0.2
future==0.18.2
greenlet==1.0.0
hkdf==0.0.3
humanize==1.0.0
hyperlink==21.0.0
hypothesis==4.57.1
idna==2.10
importlib-metadata==2.1.1
importlib-resources==3.3.1
incremental==21.3.0
ipaddress==1.0.23
Jinja2==2.11.3
linecache2==1.0.0
-e git+git@github.com:LeastAuthority/magic-folder.git@693af4662f804d33d09fb23cc9313fcdb2133070#egg=magic_folder
magic-wormhole==0.12.0
MarkupSafe==1.1.1
mock==3.0.5
more-itertools==5.0.0
Nevow==0.14.5
packaging==20.9
pathlib2==2.3.5
pbr==5.6.0
pluggy==0.13.1
py==1.10.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycparser==2.20
pyflakes==2.1.0
PyHamcrest==1.10.1
PyNaCl==1.4.0
pyOpenSSL==19.1.0
pyparsing==2.4.7
pyrsistent==0.16.1
pytest==4.6.11
pytest-twisted==1.13.2
python-mimeparse==1.6.0
pyutil==3.3.0
PyYAML==5.4.1
requests==2.25.1
scandir==1.10.0
service-identity==18.1.0
singledispatch==3.6.1
six==1.15.0
sortedcontainers==2.3.0
spake2==0.8
tahoe-lafs==1.14.0
testtools==2.4.0
toml==0.10.2
towncrier==21.3.0
tox==3.23.0
tqdm==4.60.0
traceback2==1.4.0
treq==21.1.0
Twisted==20.3.0
txaio==18.8.1
txtorcon==20.0.0
typing==3.10.0.0
unittest2==1.1.0
urllib3==1.26.4
virtualenv==20.4.4
wcwidth==0.2.5
zfec==1.5.5
zipp==1.2.0
zope.interface==5.4.0
```

And then, on master, I see this cryptography warning:

```
$ magic-folder 
/home/exarkun/VirtualEnvs/_home_exarkun_Work_python_magic-folder/lib/python2.7/site-packages/OpenSSL/crypto.py:12: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release.
  from cryptography import x509
...
```

I'm guessing this is the warning that this issue/PR is about. 

When I switch to this branch and update the venv:

```sh
$ pip freeze | grep magic-folder
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support pip 21.0 will remove support for this functionality.
-e git+git@github.com:LeastAuthority/magic-folder.git@693af4662f804d33d09fb23cc9313fcdb2133070#egg=magic_folder
(_home_exarkun_Work_python_magic-folder) 
[nix-shell:~/Work/python/magic-folder]$ git rev-parse HEAD
693af4662f804d33d09fb23cc9313fcdb2133070
(_home_exarkun_Work_python_magic-folder) 
[nix-shell:~/Work/python/magic-folder]$ 
```

The warning remains:

```sh
$ magic-folder 
/home/exarkun/VirtualEnvs/_home_exarkun_Work_python_magic-folder/lib/python2.7/site-packages/OpenSSL/crypto.py:12: CryptographyDeprecationWarning: Python 2 is no longer supported by the Python core team. Support for it is now deprecated in cryptography, and will be removed in the next release.
  from cryptography import x509
...
```

Poking around a little I see that the line number in the pyOpenSSL I have installed doesn't match the lineno filter here so I suppose pyOpenSSL lines moved around a little bit between versions.

A couple other thoughts -

* `module` is documented as a case-sensitive regular expression
* If some other module accidentally starts to import cryptography first then the warning will start showing up again

Maybe filtering on the message is the most reliable approach.

Reply:
Okay, added a `message=""` regular expression .. not sure how stable that will be either, so I didn't match on the _entire_ message...